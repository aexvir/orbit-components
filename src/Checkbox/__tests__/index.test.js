// @flow

import * as React from "react";
import { shallow } from "enzyme";

import CheckBox from "../";

const label = "CheckBox";
const onChange = jest.fn();
const name = "checkbox";
const values = ["value", ""];

values.map(value =>
  describe(`Default CheckBox with value: ${value}`, () => {
    const component = shallow(
      <CheckBox label={label} onChange={onChange} name={name} value={value} />,
    );
    it("Should contain a label", () => {
      expect(component.find(".labelText").text()).toBe(label);
    });
    it("Inputs value should match", () => {
      expect(component.find(`input[type="checkbox"]`).prop("value")).toBe(value);
    });
    it("Input should not contain a disabled prop", () => {
      expect(component.find(`input[type="checkbox"]`).prop("disabled")).toBe(false);
    });
    it("Should not execute onChange method", () => {
      component.find("input").simulate("change");
      expect(onChange).toHaveBeenCalled();
    });
    it("Should match snapshot", () => {
      expect(component).toMatchSnapshot();
    });
  }),
);

values.map(value =>
  describe(`Disabled CheckBox with value: ${value}`, () => {
    const component = shallow(
      <CheckBox label={label} onChange={onChange} name={name} value={value} disabled />,
    );
    it("Should contain a label", () => {
      expect(component.find(".labelText").text()).toBe(label);
    });
    it("Inputs value should match", () => {
      expect(component.find(`input[type="checkbox"]`).prop("value")).toBe(value);
    });
    it("Input should contain a disabled prop", () => {
      expect(component.find(`input[type="checkbox"]`).prop("disabled")).toBe(true);
    });
    it("Should match snapshot", () => {
      expect(component).toMatchSnapshot();
    });
  }),
);
