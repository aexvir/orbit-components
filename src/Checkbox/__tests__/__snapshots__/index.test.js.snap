// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default CheckBox with value:  Should match snapshot 1`] = `
<label
  className="jsx-2232527030"
  htmlFor="checkbox"
>
  <input
    checked={false}
    className="jsx-2232527030"
    disabled={false}
    id="checkbox"
    name="checkbox"
    onChange={
      [MockFunction] {
        "calls": Array [
          Array [],
          Array [],
        ],
      }
    }
    type="checkbox"
    value=""
  />
  <span
    className="jsx-2232527030 checkBox"
  >
    <span
      className="jsx-2232527030 glyph"
    >
      <Check
        className="CheckIcon"
        color="#00a991"
        size="16px"
      />
    </span>
  </span>
  <span
    className="jsx-2232527030 labelText"
  >
    CheckBox
  </span>
  <JSXStyle
    css="label.__jsx-style-dynamic-selector{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;height:20px;opacity:1;cursor:pointer;position:relative;}input[type=\\"checkbox\\"].__jsx-style-dynamic-selector{visibility:hidden;display:none;}.checkBox.__jsx-style-dynamic-selector{position:relative;box-sizing:border-box;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;background-color:#fff;height:20px;width:20px;border-radius:3px;overflow:hidden;border:1px solid #bac7d5;-webkit-transform:scale(1);-ms-transform:scale(1);transform:scale(1);-webkit-transition:all 0.2s linear;transition:all 0.2s linear;}.checkBox.__jsx-style-dynamic-selector .glyph.__jsx-style-dynamic-selector{visibility:hidden;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;}input[type=\\"checkbox\\"].__jsx-style-dynamic-selector:checked+.checkBox.__jsx-style-dynamic-selector .glyph.__jsx-style-dynamic-selector{visibility:visible;}label.__jsx-style-dynamic-selector:hover .checkBox.__jsx-style-dynamic-selector{border-color:#bac7d5;}.labelText.__jsx-style-dynamic-selector{font-family:\\"Roboto\\", -apple-system, \\".SFNSText-Regular\\", \\"San Francisco\\", \\"Segoe UI\\", \\"Helvetica Neue\\", \\"Lucida Grande\\", sans-serif;font-weight:400;font-size:14px;color:#46515e;margin-left:8px;}input[type=\\"checkbox\\"].__jsx-style-dynamic-selector:checked~.labelText.__jsx-style-dynamic-selector{font-weight:500;}label.__jsx-style-dynamic-selector:active .checkBox.__jsx-style-dynamic-selector{border-color:#9aadc2;transform: scale(0.9);}label.__jsx-style-dynamic-selector:focus{outline-width:3px;}"
    dynamic={
      Array [
        "20px",
        "1",
        "pointer",
        "#fff",
        "20px",
        "20px",
        "3px",
        "#bac7d5",
        "0.2s",
        "#bac7d5",
        "\\"Roboto\\", -apple-system, \\".SFNSText-Regular\\", \\"San Francisco\\", \\"Segoe UI\\", \\"Helvetica Neue\\", \\"Lucida Grande\\", sans-serif",
        "400",
        "14px",
        "#46515e",
        "8px",
        "500",
        "#9aadc2",
        "transform: scale(0.9)",
      ]
    }
    styleId="159042117"
  />
</label>
`;

exports[`Default CheckBox with value: value Should match snapshot 1`] = `
<label
  className="jsx-2232527030"
  htmlFor="checkbox"
>
  <input
    checked={false}
    className="jsx-2232527030"
    disabled={false}
    id="checkbox"
    name="checkbox"
    onChange={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
      }
    }
    type="checkbox"
    value="value"
  />
  <span
    className="jsx-2232527030 checkBox"
  >
    <span
      className="jsx-2232527030 glyph"
    >
      <Check
        className="CheckIcon"
        color="#00a991"
        size="16px"
      />
    </span>
  </span>
  <span
    className="jsx-2232527030 labelText"
  >
    CheckBox
  </span>
  <JSXStyle
    css="label.__jsx-style-dynamic-selector{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;height:20px;opacity:1;cursor:pointer;position:relative;}input[type=\\"checkbox\\"].__jsx-style-dynamic-selector{visibility:hidden;display:none;}.checkBox.__jsx-style-dynamic-selector{position:relative;box-sizing:border-box;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;background-color:#fff;height:20px;width:20px;border-radius:3px;overflow:hidden;border:1px solid #bac7d5;-webkit-transform:scale(1);-ms-transform:scale(1);transform:scale(1);-webkit-transition:all 0.2s linear;transition:all 0.2s linear;}.checkBox.__jsx-style-dynamic-selector .glyph.__jsx-style-dynamic-selector{visibility:hidden;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;}input[type=\\"checkbox\\"].__jsx-style-dynamic-selector:checked+.checkBox.__jsx-style-dynamic-selector .glyph.__jsx-style-dynamic-selector{visibility:visible;}label.__jsx-style-dynamic-selector:hover .checkBox.__jsx-style-dynamic-selector{border-color:#bac7d5;}.labelText.__jsx-style-dynamic-selector{font-family:\\"Roboto\\", -apple-system, \\".SFNSText-Regular\\", \\"San Francisco\\", \\"Segoe UI\\", \\"Helvetica Neue\\", \\"Lucida Grande\\", sans-serif;font-weight:400;font-size:14px;color:#46515e;margin-left:8px;}input[type=\\"checkbox\\"].__jsx-style-dynamic-selector:checked~.labelText.__jsx-style-dynamic-selector{font-weight:500;}label.__jsx-style-dynamic-selector:active .checkBox.__jsx-style-dynamic-selector{border-color:#9aadc2;transform: scale(0.9);}label.__jsx-style-dynamic-selector:focus{outline-width:3px;}"
    dynamic={
      Array [
        "20px",
        "1",
        "pointer",
        "#fff",
        "20px",
        "20px",
        "3px",
        "#bac7d5",
        "0.2s",
        "#bac7d5",
        "\\"Roboto\\", -apple-system, \\".SFNSText-Regular\\", \\"San Francisco\\", \\"Segoe UI\\", \\"Helvetica Neue\\", \\"Lucida Grande\\", sans-serif",
        "400",
        "14px",
        "#46515e",
        "8px",
        "500",
        "#9aadc2",
        "transform: scale(0.9)",
      ]
    }
    styleId="159042117"
  />
</label>
`;

exports[`Disabled CheckBox with value:  Should match snapshot 1`] = `
<label
  className="jsx-1504766170"
  htmlFor="checkbox"
>
  <input
    checked={false}
    className="jsx-1504766170"
    disabled={true}
    id="checkbox"
    name="checkbox"
    onChange={
      [MockFunction] {
        "calls": Array [
          Array [],
          Array [],
        ],
      }
    }
    type="checkbox"
    value=""
  />
  <span
    className="jsx-1504766170 checkBox"
  >
    <span
      className="jsx-1504766170 glyph"
    >
      <Check
        className="CheckIcon"
        color="#bac7d5"
        size="16px"
      />
    </span>
  </span>
  <span
    className="jsx-1504766170 labelText"
  >
    CheckBox
  </span>
  <JSXStyle
    css="label.__jsx-style-dynamic-selector{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;height:20px;opacity:0.5;cursor:default;position:relative;}input[type=\\"checkbox\\"].__jsx-style-dynamic-selector{visibility:hidden;display:none;}.checkBox.__jsx-style-dynamic-selector{position:relative;box-sizing:border-box;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;background-color:#fff;height:20px;width:20px;border-radius:3px;overflow:hidden;border:1px solid #bac7d5;-webkit-transform:scale(1);-ms-transform:scale(1);transform:scale(1);-webkit-transition:all 0.2s linear;transition:all 0.2s linear;}.checkBox.__jsx-style-dynamic-selector .glyph.__jsx-style-dynamic-selector{visibility:hidden;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;}input[type=\\"checkbox\\"].__jsx-style-dynamic-selector:checked+.checkBox.__jsx-style-dynamic-selector .glyph.__jsx-style-dynamic-selector{visibility:visible;}label.__jsx-style-dynamic-selector:hover .checkBox.__jsx-style-dynamic-selector{border-color:#aabacb;}.labelText.__jsx-style-dynamic-selector{font-family:\\"Roboto\\", -apple-system, \\".SFNSText-Regular\\", \\"San Francisco\\", \\"Segoe UI\\", \\"Helvetica Neue\\", \\"Lucida Grande\\", sans-serif;font-weight:400;font-size:14px;color:#46515e;margin-left:8px;}input[type=\\"checkbox\\"].__jsx-style-dynamic-selector:checked~.labelText.__jsx-style-dynamic-selector{font-weight:500;}label.__jsx-style-dynamic-selector:active .checkBox.__jsx-style-dynamic-selector{border-color:#bac7d5;;}label.__jsx-style-dynamic-selector:focus{outline-width:3px;}"
    dynamic={
      Array [
        "20px",
        "0.5",
        "default",
        "#fff",
        "20px",
        "20px",
        "3px",
        "#bac7d5",
        "0.2s",
        "#aabacb",
        "\\"Roboto\\", -apple-system, \\".SFNSText-Regular\\", \\"San Francisco\\", \\"Segoe UI\\", \\"Helvetica Neue\\", \\"Lucida Grande\\", sans-serif",
        "400",
        "14px",
        "#46515e",
        "8px",
        "500",
        "#bac7d5",
        "",
      ]
    }
    styleId="159042117"
  />
</label>
`;

exports[`Disabled CheckBox with value: value Should match snapshot 1`] = `
<label
  className="jsx-1504766170"
  htmlFor="checkbox"
>
  <input
    checked={false}
    className="jsx-1504766170"
    disabled={true}
    id="checkbox"
    name="checkbox"
    onChange={
      [MockFunction] {
        "calls": Array [
          Array [],
          Array [],
        ],
      }
    }
    type="checkbox"
    value="value"
  />
  <span
    className="jsx-1504766170 checkBox"
  >
    <span
      className="jsx-1504766170 glyph"
    >
      <Check
        className="CheckIcon"
        color="#bac7d5"
        size="16px"
      />
    </span>
  </span>
  <span
    className="jsx-1504766170 labelText"
  >
    CheckBox
  </span>
  <JSXStyle
    css="label.__jsx-style-dynamic-selector{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;height:20px;opacity:0.5;cursor:default;position:relative;}input[type=\\"checkbox\\"].__jsx-style-dynamic-selector{visibility:hidden;display:none;}.checkBox.__jsx-style-dynamic-selector{position:relative;box-sizing:border-box;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;background-color:#fff;height:20px;width:20px;border-radius:3px;overflow:hidden;border:1px solid #bac7d5;-webkit-transform:scale(1);-ms-transform:scale(1);transform:scale(1);-webkit-transition:all 0.2s linear;transition:all 0.2s linear;}.checkBox.__jsx-style-dynamic-selector .glyph.__jsx-style-dynamic-selector{visibility:hidden;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;}input[type=\\"checkbox\\"].__jsx-style-dynamic-selector:checked+.checkBox.__jsx-style-dynamic-selector .glyph.__jsx-style-dynamic-selector{visibility:visible;}label.__jsx-style-dynamic-selector:hover .checkBox.__jsx-style-dynamic-selector{border-color:#aabacb;}.labelText.__jsx-style-dynamic-selector{font-family:\\"Roboto\\", -apple-system, \\".SFNSText-Regular\\", \\"San Francisco\\", \\"Segoe UI\\", \\"Helvetica Neue\\", \\"Lucida Grande\\", sans-serif;font-weight:400;font-size:14px;color:#46515e;margin-left:8px;}input[type=\\"checkbox\\"].__jsx-style-dynamic-selector:checked~.labelText.__jsx-style-dynamic-selector{font-weight:500;}label.__jsx-style-dynamic-selector:active .checkBox.__jsx-style-dynamic-selector{border-color:#bac7d5;;}label.__jsx-style-dynamic-selector:focus{outline-width:3px;}"
    dynamic={
      Array [
        "20px",
        "0.5",
        "default",
        "#fff",
        "20px",
        "20px",
        "3px",
        "#bac7d5",
        "0.2s",
        "#aabacb",
        "\\"Roboto\\", -apple-system, \\".SFNSText-Regular\\", \\"San Francisco\\", \\"Segoe UI\\", \\"Helvetica Neue\\", \\"Lucida Grande\\", sans-serif",
        "400",
        "14px",
        "#46515e",
        "8px",
        "500",
        "#bac7d5",
        "",
      ]
    }
    styleId="159042117"
  />
</label>
`;
